NEWS.md

```diff
@@ -359,9 +359,15 @@ This section lists changes that do not have deprecation warnings.
     trait; see its documentation for details. Types which support subtraction (operator
     `-`) must now implement `widen` for hashing to work inside heterogeneous arrays.
 
-  * `findn(x::AbstractVector)` now return a 1-tuple with the vector of indices, to be
+  * `findn(x::AbstractVector)` now returns a 1-tuple with the vector of indices, to be
     consistent with higher order arrays ([#25365]).
 
+  * `find` now returns the same type of indices as `keys`/`pairs` for `AbstractArray` and
+    `AbstractDict` objects ([#24774]). In particular, this means that it returns
+    `CartesianIndex` objects for matrices and higher-dimensional arrays instead of
+    always returning linear indices as it was previously the case. Use
+    `Int[LinearIndices(size(a))[i] for i in find(f, a)]` to compute linear indices
```

Comment:
@timholy Is there any reason why you cannot index `LinearIndices` objects with an array of cartesian indices? That would provide a much more convenient syntax.

Though as I noted in my last comment it would make sense to provide a `find` method so that no temporary allocation is needed.

Reply:
If I understand your question correctly, that's possible because
```julia
julia> LinearIndices <: AbstractArray
true
```
Here's a demo of what I mean:
```julia
julia> linear = LinearIndices(1:3, 1:5)
LinearIndices{2,Tuple{UnitRange{Int64},UnitRange{Int64}}} with indices 1:3×1:5:
 1  4  7  10  13
 2  5  8  11  14
 3  6  9  12  15

julia> linear[3]
3

julia> linear[3,2]
6

julia> linear[[CartesianIndex(3,2), CartesianIndex(2,3)]]
2-element Array{Int64,1}:
 6
 8
```

Reply:
Hmm, I wonder why I thought it didn't work... So everything is fine.

While I'm at it, should we also support `linear[[1, 2]]`? That would be convenient for backward compatibility, since you would be able to apply this operation to indices returned by `find`, and it would be a no-op on 0.6. It would also be more consistent.

Reply:
We have that already, too :smile:. But it currently fails because we seem to be missing a `size` method for `LinearIndices`.

Reply:
See https://github.com/JuliaLang/julia/pull/25541.